ksqldb-engine/src/main/java/io/confluent/ksql/function/udf/array/ArrayJoin.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.function.udf.array;
+
+import io.confluent.ksql.function.KsqlFunctionException;
+import io.confluent.ksql.function.udf.Udf;
+import io.confluent.ksql.function.udf.UdfDescription;
+import io.confluent.ksql.function.udf.UdfParameter;
+import io.confluent.ksql.util.KsqlConstants;
+import java.math.BigInteger;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringJoiner;
+import org.apache.kafka.connect.data.Struct;
+
+@SuppressWarnings("MethodMayBeStatic") // UDF methods can not be static.
+@UdfDescription(
+    name = "ARRAY_JOIN",
+    description = "joins the array elements into a flat string representation",
+    author = KsqlConstants.CONFLUENT_AUTHOR
+)
+public class ArrayJoin {
+
+  private static final String DEFAULT_DELIMITER = ",";
+  private static final Set<Class> KSQL_PRIMITIVES = new HashSet<>(
+      Arrays.asList(Boolean.class,Integer.class,BigInteger.class,Double.class,String.class)
```

Comment:
why `BigInteger`? I think ksqlDB only uses `BigDecimal` internally - there is also `Long.class` if you are referring to the sql `BIGINT` type

Reply:
fair point. I guess I mistakenly assumed that sql `BIGINT` would map to `BigInteger` which obviously is not the case? Should I switch to `Long` cause `BigDecimal` seems even weirder to me than `BigInteger` when thinking about it.

Reply:
I was a little unclear; we should remove `BigInteger` and add both `Long` and `BigDecimal`. `BigDecimal` is the java class that we use to represent the sql `DECIMAL` type, which should be supported in this method

Reply:
Thx for clarifying. The reason I didn't consider DECIMAL is because it simply didn't work for me. Maybe I'm too stupid but trying to use the ksql CLI and create an `ARRAY` of `DECIMAL`s   never worked for me thus I thought it's currently not supported :)

Reply:
Not stupid, haha - our UX isn't great with decimals, but the below should work:
```sql
ksql> CREATE STREAM dec_array (val ARRAY<DECIMAL(2,1)>) WITH (kafka_topic='foo', value_format='json', partitions=1);

 Message
----------------
 Stream created
----------------
ksql> INSERT INTO dec_array (val) VALUES (ARRAY[2.1]);
ksql> SELECT * FROM dec_array EMIT CHANGES;
+----------------------------------------------+----------------------------------------------+
|ROWKEY                                        |VAL                                           |
+----------------------------------------------+----------------------------------------------+
|null                                          |[2.1]                                         |
```

Reply:
done!